templates/tfengine/components/project/bigquery_datasets/main.tf

```diff
@@ -0,0 +1,32 @@
+{{- /* Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */ -}}
+
+{{range get . "BIGQUERY_DATASETS"}}
+module "{{resourceName .DATASET_ID}}" {
+  source  = "terraform-google-modules/bigquery/google"
+  version = "~> 4.2.0"
+
+  dataset_id = "{{.DATASET_ID}}"
+  project_id = var.project_id
+  location   = "{{get . "LOCATION" $.BIGQUERY_DATASET_LOCATION}}"
+
+  {{- if has . "DEFAULT_TABLE_EXPIRATION_MS"}}
+  default_table_expiration_ms = {{.DEFAULT_TABLE_EXPIRATION_MS}}
+  {{- end}}
+
+  {{if has . "ACCESS"}}
```

Comment:
Mentioned it offline before, but I think this block can be simplified to:
```
{{hcl_has . "ACCESS" "access"}}
```
Where `hcl_has` would either return `"<field> = hclencoder.Encode(v)"` or `""` (nothing) if `has` fails.

And same with other blocks.

Reply:
Done